WordPress/src/main/java/org/wordpress/android/ui/activitylog/ActivityLogDetailFragment.kt

```diff
@@ -0,0 +1,111 @@
+package org.wordpress.android.ui.activitylog
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProvider
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import kotlinx.android.synthetic.main.activity_log_item_detail.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.viewmodel.activitylog.ACTIVITY_LOG_ID_KEY
+import org.wordpress.android.viewmodel.activitylog.ActivityLogDetailViewModel
+import org.wordpress.android.widgets.WPNetworkImageView
+import javax.inject.Inject
+
+class ActivityLogDetailFragment : Fragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: ActivityLogDetailViewModel
+
+    companion object {
+        fun newInstance(): ActivityLogDetailFragment {
+            return ActivityLogDetailFragment()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (activity?.application as WordPress).component()?.inject(this)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
+                .get<ActivityLogDetailViewModel>(ActivityLogDetailViewModel::class.java)
+
+        val intent = activity?.intent
+        val (site, activityLogId) = when {
+            savedInstanceState != null -> {
+                val site = savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+                val activityLogId = savedInstanceState.getString(ACTIVITY_LOG_ID_KEY)
+                site to activityLogId
+            }
+            intent != null -> {
+                val site = intent.getSerializableExtra(WordPress.SITE) as SiteModel
+                val activityLogId = intent.getStringExtra(ACTIVITY_LOG_ID_KEY)
+                site to activityLogId
+            }
+            else -> throw Throwable("Couldn't initialize Activity Log view model")
+        }
+        viewModel.activityLogItem.observe(this, Observer { activityLogModel ->
+            setActorIcon(activityLogModel?.actorIconUrl)
+            activity_actor_name.setTextOrHide(activityLogModel?.actorName)
```

Comment:
Since we're using generated variables, what do you think about using CamelCase for view names in the layout?

Reply:
I don't have a strong opinion about this so why not 👍